### PR TITLE
Remove StreamerInfo `EReadWrite` duplication

### PR DESCRIPTION
**Pull request by pcanal**
Also add 'missing' `EReadWrite` values.

### CODE REVIEW

File: io/io/src/TStreamerInfo.cxx [hunk 1]
```diff
@@ -2216,7 +2218,7 @@ void TStreamerInfo::BuildOld()
          // Case of a numerical type
          if (element->GetType() >= TStreamerInfo::kObject) {
             // Old type was not a numerical type.
-            element->SetNewType(-2);
+            element->SetNewType(kUnsupportedConversion);
```
Comment (vepadulano) [inline io/io/src/TStreamerInfo.cxx:2221]: Why is this missing the `TVirtualStreamerInfo::`?

File: core/meta/src/TStreamerElement.cxx [hunk 2]
```diff
@@ -195,9 +195,9 @@ ClassImp(TStreamerElement);
 
 TStreamerElement::TStreamerElement()
 {
-   fType        = 0;
+   fType        = TVirtualStreamerInfo::kUnset;
```
Comment (vepadulano) [inline core/meta/src/TStreamerElement.cxx:198]: IIUC here the value is changed from `0` to `-3`, I believe this is a conscious choice but I wanted to make sure that we know it won't lead to unintended behaviour elsewhere (e.g. some function expecting `fType == 0`).

**Review by vepadulano — COMMENTED** [review]
Some comments for clarification

File: io/io/src/TStreamerInfo.cxx (hunk 1 above)
Comment (pcanal) [reply io/io/src/TStreamerInfo.cxx:2221]: Oversight :(

**Review by vepadulano — APPROVED** [review]
Thanks! Consider applying the changes suggested by clang-format before merging

**Comment by pcanal** [comment]
> Consider applying the changes suggested by clang-format before merging

Sure but 90% of those were 'worse', so I disable clang format around the related lines.

**Pull request merged**